File: src/main/java/ru/citeck/ecos/records2/graphql/meta/value/factory/DateValueFactory.java
package ru.citeck.ecos.records2.graphql.meta.value.factory;

import com.fasterxml.jackson.databind.util.ISO8601Utils;
import ru.citeck.ecos.records2.graphql.meta.value.MetaValue;

import java.util.Collections;
import java.util.Date;
import java.util.List;

public class DateValueFactory implements MetaValueFactory<Date> {

    @Override
    public MetaValue getValue(Date value) {
        return new DateValue(value);
    }

    @Override
    public List<Class<? extends Date>> getValueTypes() {
        return Collections.singletonList(Date.class);
    }

    static class DateValue implements MetaValue {

        private Date date;

        public DateValue(Date date) {
            this.date = date;
        }

        @Override
        public String getString() {
            return ISO8601Utils.format(date);
        }
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/service/value/factory/time/DateValueFactory.kt
package ru.citeck.ecos.records3.record.op.atts.service.value.factory.time

import ru.citeck.ecos.records3.record.op.atts.service.value.AttValue
import ru.citeck.ecos.records3.record.op.atts.service.value.factory.AttValueFactory
import java.time.Instant
import java.util.*

class DateValueFactory : AttValueFactory<Date> {

    override fun getValue(value: Date): AttValue? {
        return DateValue(value)
    }

    override fun getValueTypes() = listOf(Date::class.java)

    class DateValue(private val date: Date) : AttValue {

        override fun asText(): String {
            return Instant.ofEpochMilli(date.time).toString()
        }

        override fun asDouble(): Double? {
            return date.time.toDouble()
        }
    }
}
